arwen/context/arwen.go

```diff
@@ -804,6 +840,8 @@ func (host *vmContext) execute(input *vmcommon.ContractCallInput) error {
 	convertedResult := arwen.ConvertReturnValue(result)
```

Comment:
Potential gas leak: my malicious SmartContract could run very expensive code, then trigger a `signalError()`, which will set `host.returnCode` to `vmcommon.UserError`. The lines above (see `if host.returnCode != vmcommon.Ok {...}`) do not deduct gas if this happens.